### PR TITLE
Refactor HttpRequest to support body as array of bytes

### PR DESCRIPTION
**Pull request by snahornyi**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/model/HttpRequest.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
     String uri;
 
-    String body;
+    byte[] body;
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/model/HttpRequest.java:19]: Minor. Pls move this field down to be set after `headers`.

File: src/main/java/org/prebid/server/bidder/BidderErrorNotifier.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ private Void handleTimeoutNotificationSuccess(HttpClientResponse response, HttpR
                     String.format(
                             "Notified bidder about timeout. Status code: %s. Request body: %s",
                             response.getStatusCode(),
-                            timeoutNotification.getBody()),
+                            new String(timeoutNotification.getBody())),
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/BidderErrorNotifier.java:72]: Why we need this?

File: src/main/java/org/prebid/server/bidder/algorix/AlgorixBidder.java [hunk 3]
```diff
@@ -74,7 +74,7 @@ public AlgorixBidder(String endpointUrl, JacksonMapper mapper) {
         }
 
         final BidRequest outgoingRequest = request.toBuilder().imp(updatedImps).build();
-        final String body = mapper.encode(outgoingRequest);
+        final byte[] body = mapper.encodeToBytes(outgoingRequest);
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/algorix/AlgorixBidder.java:77]: Pls inline the variable.

File: src/main/java/org/prebid/server/bidder/applogy/ApplogyBidder.java [hunk 4]
```diff
@@ -68,7 +68,7 @@ public ApplogyBidder(String endpointUrl, JacksonMapper mapper) {
     private HttpRequest<BidRequest> createSingleRequest(Imp imp, BidRequest request, String url) {
         final BidRequest outgoingRequest = request.toBuilder().imp(Collections.singletonList(imp)).build();
 
-        final String body = mapper.encode(outgoingRequest);
+        final byte[] body = mapper.encodeToBytes(outgoingRequest);
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/applogy/ApplogyBidder.java:71]: Pls inline the variable.

File: src/main/java/org/prebid/server/bidder/avocet/AvocetBidder.java [hunk 5]
```diff
@@ -42,9 +42,9 @@ public AvocetBidder(String endpointUrl, JacksonMapper mapper) {
 
     @Override
     public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
-        final String body;
+        final byte[] body;
         try {
-            body = mapper.encode(request);
+            body = mapper.encodeToBytes(request);
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/avocet/AvocetBidder.java:47]: 1. Pls inline the variable.
2. i'd say it's not required to be safe here, pls remove `try-catch`.

File: src/main/java/org/prebid/server/bidder/brightroll/BrightrollBidder.java [hunk 6]
```diff
@@ -77,9 +77,9 @@ public BrightrollBidder(String endpointUrl,
             return Result.withErrors(errors);
         }
 
-        final String bidRequestBody;
+        final byte[] bidRequestBody;
         try {
-            bidRequestBody = mapper.encode(updateBidRequest);
+            bidRequestBody = mapper.encodeToBytes(updateBidRequest);
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/brightroll/BrightrollBidder.java:82]: Same as above.

File: src/main/java/org/prebid/server/bidder/connectad/ConnectadBidder.java [hunk 7]
```diff
@@ -72,7 +72,7 @@ public ConnectadBidder(String endpointUrl, JacksonMapper mapper) {
         }
         final BidRequest outgoingRequest = request.toBuilder().imp(processedImps).build();
 
-        final String body = mapper.encode(outgoingRequest);
+        final byte[] body = mapper.encodeToBytes(outgoingRequest);
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/connectad/ConnectadBidder.java:75]: Pls inline the variable.

File: src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java [hunk 8]
```diff
@@ -95,9 +95,9 @@ public ConsumableBidder(String endpointUrl, JacksonMapper mapper) {
         }
 
         final ConsumableBidRequest outgoingRequest = requestBuilder.build();
-        final String body;
+        final byte[] body;
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java:98]: Pls inline the variable.

File: src/main/java/org/prebid/server/bidder/kubient/KubientBidder.java [hunk 9]
```diff
@@ -55,9 +55,9 @@ public KubientBidder(String endpointUrl, JacksonMapper mapper) {
             }
         }
 
-        String body;
+        final byte[] body;
```
Comment (rpanchyk) [inline src/main/java/org/prebid/server/bidder/kubient/KubientBidder.java:58]: Pls inline the variable.

**Review by rpanchyk — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/BidderErrorNotifier.java (hunk 2 above)
Comment (snahornyi) [reply src/main/java/org/prebid/server/bidder/BidderErrorNotifier.java:72]: Screenshot 2021-11-09 at 11 13 18

**Review by rpanchyk — APPROVED** [review]

File: src/main/java/org/prebid/server/bidder/BidderErrorNotifier.java (hunk 2 above)
Comment (rpanchyk) [reply src/main/java/org/prebid/server/bidder/BidderErrorNotifier.java:72]: Agreed.

**Pull request merged**